chain/client/src/info.rs

```diff
@@ -142,10 +142,14 @@ impl InfoHelper {
     }
 
     /// Count which shards are tracked by the node in the epoch indicated by head parameter.
-    fn record_tracked_shards(head: &Tip, client: &crate::client::Client) {
-        let validator_signer = client.validator_signer.get();
-        let me = validator_signer.as_ref().map(|x| x.validator_id());
-        if let Ok(shard_ids) = client.epoch_manager.shard_ids(&head.epoch_id) {
+    fn record_tracked_shards(
+        head: &Tip,
+        client: &crate::client::Client,
+        shard_ids: Option<Vec<ShardId>>,
```

Comment:
nit: to me it looks a bit weird to see a function that accepts an optional arg and does nothing if it is `None`, I think it would be cleaner to make it `shard_ids: &[ShardId]` and check for `None` on the caller side

Reply:
+1 And it also feels like this should always be the list of all shard ids at the given epoch. Assuming this is the case it would be better to pass the shard layout.

Reply:
yeah, changing it to ShardLayout, since there is also a problem with the chunk validator stats using a subset of the shard ids, so using ShardLayout should be cleaner than a vector of shard ids.